src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,122 @@
+package core.basesyntax.service;
+
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+    private static RegistrationServiceImpl registrationService;
+    private static StorageDaoImpl storageDao;
+
+    @BeforeAll
+    static void beforeAll() {
+        registrationService = new RegistrationServiceImpl();
+        storageDao = new StorageDaoImpl();
+    }
+
+    @BeforeEach
+    void setUp() {
+        storageDao.add(new User(123456789L, "unknownCactus", "CanLive_withoutWater22", 23));
+        storageDao.add(new User(234567891L, "notASportsman", "KFC_Matters121212", 45));
+        storageDao.add(new User(345678912L, "starInTheSky", "needATelescopeToSeeMe777", 18));
+        storageDao.add(new User(456789123L, "baldBlueBeard", "ShutUP_andTakeMyMoney$$", 57));
+    }
+
+    @Test

```

Comment:
your user don`t need to initialize field id 
field id it`s just counter for arrayList